fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/importhandler/client/ClientEntityImportHandler.java

```diff
@@ -141,7 +143,7 @@ private ClientData readClient(final Workbook workbook, final Row row, final Stri
         ClientNonPersonData clientNonPersonData = ClientNonPersonData.importInstance(incorporationNo, incorporationTill, remarks,
                 mainBusinessId, constitutionId, locale, dateFormat);
 
-        String externalId = ImportHandlerUtils.readAsString(ClientEntityConstants.EXTERNAL_ID_COL, row);
+        ExternalId externalId = ExternalIdFactory.produce(ImportHandlerUtils.readAsString(ClientEntityConstants.EXTERNAL_ID_COL, row));
```

Comment:
Would it make more sense to use ExternalIdFactory.create()? it would auto generate one if the configuration is enabled and value was not provided

Reply:
Done!